src/services/item/plugins/invitation/service.ts

```diff
@@ -0,0 +1,350 @@
+      return { permission, memberId: account.id };
+    });
+    this.log.debug(membershipsToCreate, 'memberships to create');
+
+    // create memberships for accounts that already exist
+    const memberships = await itemMembershipService.postMany(
+      actor,
+      repositories,
+      membershipsToCreate,
+      itemId,
+    );
+
+    // generate invitations to create
+    const invitationsToCreate = newAccounts.map((email) => {
+      // get the permission from the data, if it is not found or if it is an empty string, default to read
+      const permission = rows.find((r) => r.email === email)?.permission || PermissionLevel.Read;
+      return { email, permission };
+    });
+    this.log.debug(invitationsToCreate, 'invitations to create');
+
+    // create invitations for accounts that do not exist yet
+    const invitations = await this.postManyForItem(
+      actor,
+      repositories,
+      itemId,
+      invitationsToCreate,
+    );
+
+    return { memberships, invitations };
+  }
+
+  async importUsersWithCSV(
+    actor: Member,
+    repositories: Repositories,
+    itemId: Item['id'],
+    file: MultipartFile,
+    itemMembershipService: ItemMembershipService,
+  ): Promise<{ memberships: ItemMembership[]; invitations: Invitation[] }> {
+    // verify file is CSV
+    verifyCSVFileFormat(file);
+
+    // get parentItem
+    const item = await repositories.itemRepository.get(itemId);
+    await validatePermission(repositories, PermissionLevel.Admin, actor, item);
+
+    // parse CSV file
+    const { rows, header } = await parseCSV(file.file);
+
+    // check that file has data
+    if (rows.length === 0) {
+      throw new NoDataInFile();
+    }
+
+    // check that the email column has been detected
+    if (!header.includes(EMAIL_COLUMN_NAME)) {
+      throw new MissingEmailColumnInCSVError();
+    }
+
+    // check that all rows have an email set
+    if (rows.some((r) => !Boolean(r.email))) {
+      throw new MissingEmailInRowError();
+    }
+
+    return this._createMembershipsAndInvitationsForUserList(
+      actor,
+      repositories,
+      rows,
+      itemId,
+      itemMembershipService,
+    );
+  }
+
+  async handleCSVInvitations(
+    actor: Member,
+    repositories: Repositories,
+    parentId: string,
+    templateId: string,
+    file: MultipartFile,
+    itemMembershipService: ItemMembershipService,
+  ): Promise<
+    {
+      groupName: string;
+      memberships: ItemMembership[];
+      invitations: Invitation[];
+    }[]
+  > {
+    // verify file is CSV
+    verifyCSVFileFormat(file);
+
+    // get parentItem
+    const parentItem = await repositories.itemRepository.get(parentId);
+    await validatePermission(repositories, PermissionLevel.Admin, actor, parentItem);
+
+    // check that the template exists
+    const templateItem = await repositories.itemRepository.get(templateId);
+    if (!templateItem.id) {
+      throw new TemplateItemDoesNotExist();
+    }
```

Comment:
use `itemService`? This will also valide the permission.

Reply:
do you always need a template id ? the name of the function did not mention it.

Reply:
Yes this function expects a template Id, as it is the one that creates the structure from the csv, i have renamed the function to make that clear.

Reply:
And I will use the itemService instead.